pkgs/stdenv/generic/setup.sh

```diff
@@ -602,9 +603,11 @@ fi
 
 PATH="${_PATH-}${_PATH:+${PATH:+:}}$PATH"
 HOST_PATH="${_HOST_PATH-}${_HOST_PATH:+${HOST_PATH:+:}}$HOST_PATH"
+XDG_DATA_DIRS="${_XDG_DATA_DIRS-}${_XDG_DATA_DIRS:+${XDG_DATA_DIRS:+:}}${XDG_DATA_DIRS-}"
```

Comment:
I spent a couple minutes looking through Nixpkgs to see if this could interfere with anything. I don't think it will, but setting something like this can always have unintended consequences. The main concern would be if native XDG_DATA_DIRS gets mixed in with target XDG_DATA_DIRS in cross-compilation. For things like bash completion and man pages, it's not such a big deal, but XDG_DATA_DIRS is also used to locate qt5 plugins.

Reply:
Since it's not exported, this might not be as much of a problem actually.

Reply:
It is implicitly exported when `XDG_DATA_DIRS` already exists as an environment variable (and `--pure` isn't used).

Always exporting it seems more consistent. If mixing with the `nix-shell` host environment is problematic, that can be avoided with `nix-shell --pure`.

In cross compilation without `strictDeps`, mixing will occur. Perhaps `XDG_DATA_DIRS` should follow "strictDeps" behavior regardless. That will require development shells to put tools in `nativeBuildInputs` instead of `buildInputs` for bash completion to work. `nix-shell -p` gets that wrong, so it won't work there yet, but can be fixed.